app/producers/application_producer.rb

```diff
@@ -27,15 +27,27 @@ def logger
     end
 
     def kafka_ca_cert
-      return unless Settings.kafka.security_protocol == 'ssl'
+      return unless %w[ssl sasl_ssl].include?(Settings.kafka.security_protocol)
 
       File.read(Settings.kafka.ssl_ca_location)
     end
 
+    def sasl_config
+      return {} unless Settings.kafka.security_protocol == 'sasl_ssl'
+
+      {
+        sasl_scram_username: Settings.kafka.sasl_username,
+        sasl_scram_password: Settings.kafka.sasl_password,
+        sasl_scram_mechanism: 'sha512'
+      }
+    end
+
     def kafka_config
       {}.tap do |config|
         config[:client_id] = self::CLIENT_ID
         config[:ssl_ca_cert] = kafka_ca_cert if kafka_ca_cert
+
+        config.merge!(sasl_config)
```

Comment:
I'd move the condition somehow here, just as the line above has the modifier if.

Reply:
Yes, that was my initial take. :slightly_smiling_face: